libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -624,39 +624,36 @@ - (void)setOauthClientId:(NSString *)newClientId
 
 + (void)resetSessionCookie
 {
-    [self removeCookies:@[@"sid"]
-            fromDomains:@[@".salesforce.com", @".force.com", @".cloudforce.com"]];
-    [self addSidCookieForInstance];
+    __weak typeof(self) weakSelf = self;
```

Comment:
True, true. I was narrowly thinking of that sso bug which got all that work started (and which involved left over cookies in the oauth webview).